apps/api/src/view-models/buildViewModelForResource/viewModels/types/TreeViewModel.ts

```diff
@@ -0,0 +1,91 @@
+import { Book } from '../../../../domain/models/book/entities/book.entity';
+import { ResourceType, resourceTypes } from '../../../../domain/types/resourceTypes';
+import { BaseViewModel } from '../base.view-model';
+import { BookViewModel } from '../book.view-model';
+import { TagViewModel } from '../tag.view-model';
+import { ResourceTypeToViewModel } from './ResourceTypeToViewModel';
+
+export type TreeViewModel<TResourceType extends ResourceType> = {
+    resourceType: TResourceType;
+
+    nodes: ResourceTree<ResourceTypeToViewModel[TResourceType]>[];
+};
+
+type ResourceTree<TViewModel extends BaseViewModel> = {
+    tag: TagViewModel;
+
+    resources: TViewModel[];
+
+    children: ResourceTree<TViewModel>[];
+};
+
+const foo: TreeViewModel<typeof resourceTypes.book> = {
```

Comment:
Maybe a more precise name here would be helpful? :P

Reply:
But this is just a test piece of data to illustrate a concrete example of the more abstract type.